src/SSVOpenHexagon/Components/CPlayer.cpp

```diff
@@ -136,17 +127,70 @@ void CPlayer::drawDeathEffect(HexagonGame& mHexagonGame)
     }
 }
 
-void CPlayer::playerSwap(HexagonGame& mHexagonGame, bool mPlaySound)
-{
+void CPlayer::playerSwap(HexagonGame &mHexagonGame, bool mPlaySound) {
     angle += ssvu::pi;
     mHexagonGame.runLuaFunctionIfExists<void>("onCursorSwap"); // TODO: docs
 
-    if(mPlaySound)
-    {
+    if (mPlaySound) {
         mHexagonGame.getAssets().playSound("swap.ogg");
     }
 }
 
+[[nodiscard]] sf::Vector2f CPlayer::getPosition() const {
+    return pos;
+}
+
+[[nodiscard]] sf::Vector2f CPlayer::getStartPosition() const {
+    return startPos;
+}
+
+void CPlayer::kill(HexagonGame &mHexagonGame, bool mEffect) {
+    deadEffectTimer.restart();
+
+    if (!Config::getInvincible()) { dead = true; }
+    if (mEffect) { mHexagonGame.death(); }
+
+    ssvs::moveTowards(lastPos, ssvs::zeroVec2f, 5 * mHexagonGame.getSpeedMultDM());
+
+    pos = lastPos;
+}
+
+void CPlayer::push(HexagonGame& mHexagonGame, CWall& wall){
+    const auto curveData = wall.getCurve();
+    const int curveDir = ssvu::getSign(curveData.speed);
+    const int movement{mHexagonGame.getInputMovement()};
+
+    const float radius{mHexagonGame.getRadius()};
+
+    unsigned int maxAttempts;
+    if (curveDir != 0){
+        maxAttempts = std::abs(curveData.speed+speed*(movement*curveDir));
+    } else {
+        maxAttempts = speed;
+    }
+    unsigned int attempt = 0;
+    float pushDir;
+    if (curveDir != 0){
+        const int reverseDir = ((movement != 0) ? -movement : 1)*curveDir;
+        pushDir = ssvu::toRad(1.f)*curveDir*reverseDir;
+    } else {
+        pushDir = ssvu::toRad(1.f) * -movement;
+    }
```

Comment:
`const` would be nice here as well, but the logic is too complicated to put it in a ternary operator. It would be unreadable. We can use the IIFE pattern (immediately-invoked function expression):

```suggestion
    const float pushDir = [&]
    {
        if (curveDir == 0) {
            return ssvu::toRad(1.f) * -movement;
        }
       
        const int reverseDir = ((movement != 0) ? -movement : 1)*curveDir;
        return ssvu::toRad(1.f)*curveDir*reverseDir;
    }();
```

We are basically creating a lambda (anonymous function) and then immediately invoking it.